server/api/routes/subscriptions.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const { createSubscription } = require('../actions/subscriptions');
+
+// Create a new router object, where we will have CRUD routes for our subscriptions table 
+// under the route /users/:userId/subscriptions/
+const router = express.Router();
+
+// Read all user's subscriptions
+// Route: /users/:userId/subscriptions
+router.get(async (req, res) => {
+  try {
+    const data = await getSubscriptionsByUser();
+    res.status(200).json(data);
+  } catch(error) {
+    res.status(400).json(error);
+  }
+});
+
+// READ a single user subscription
+// Route: /users/:userId/subscriptions/:subscriptionId
+router.get(async (req, res) => {
+  try {
+    // execute db query getSubscriptions.
+    const data = await getSubscriptionById();
+    // When response has returned with data from the await
+    // Issue a status 200, and place data in body as JSON 
+    res.status(200).json(data);
+  } catch(error) {
+    // If query encounters an error, 
+    // Issue a 400 status, and place error message in body as JSON 
+    res.status(400).json(error)
+  }
+});
+
+// CREATE a subscription
+// Route: /users/:userId/subscriptions/new
+router.post('/new', async (req, res) => {
+  // req.body will contain the information necessary to make a new subscription
+  try {
+    const data = await createSubscription(req.body);
+    res.status(200).json(data);
+  } catch(error) {
+    res.status(400).json(error);
+  }
+});
+
+// Update a subscription
+// Route: /users/:userId/:subscriptionsId
+router.put(async (req, res) => {
```

Comment:
usually a PUT route is for insertion, and so instead of using a `'/new'` route this should instead just be the `.put('/')` route but in this scenario I'd just do a regular `.post('/')` route, and the updating route should be a `.patch('/')` route instead